src/main/kotlin/net/atos/zac/enkelvoudiginformatieobject/EnkelvoudigInformatieObjectLockService.kt

```diff
@@ -0,0 +1,81 @@
+package net.atos.zac.enkelvoudiginformatieobject
+
+import jakarta.enterprise.context.ApplicationScoped
+import jakarta.inject.Inject
+import jakarta.persistence.EntityManager
+import jakarta.persistence.PersistenceContext
+import jakarta.transaction.Transactional
+import net.atos.client.zgw.drc.DrcClientService
+import net.atos.client.zgw.zrc.ZRCClientService
+import net.atos.client.zgw.zrc.model.Zaak
+import net.atos.zac.enkelvoudiginformatieobject.model.EnkelvoudigInformatieObjectLock
+import net.atos.zac.util.UriUtil
+import nl.lifely.zac.util.AllOpen
+import nl.lifely.zac.util.NoArgConstructor
+import java.util.Optional
+import java.util.UUID
+
+@ApplicationScoped
+@Transactional
+@AllOpen
+@NoArgConstructor
+class EnkelvoudigInformatieObjectLockService @Inject constructor(
+    private val drcClientService: DrcClientService,
+    private val zrcClientService: ZRCClientService
+) {
+    @PersistenceContext(unitName = "ZaakafhandelcomponentPU")
+    private lateinit var entityManager: EntityManager
+
+    fun createLock(enkelvoudiginformatieobjectUUID: UUID, idUser: String): EnkelvoudigInformatieObjectLock =
+        EnkelvoudigInformatieObjectLock().apply {
+            this.enkelvoudiginformatieobjectUUID = enkelvoudiginformatieobjectUUID
+            userId = idUser
+            lock = drcClientService.lockEnkelvoudigInformatieobject(enkelvoudiginformatieobjectUUID)
+            entityManager.persist(this)
+        }
+
+    fun findLock(enkelvoudiginformatieobjectUUID: UUID?): Optional<EnkelvoudigInformatieObjectLock> {
```

Comment:
```suggestion
    fun findLock(enkelvoudiginformatieobjectUUID: UUID): Optional<EnkelvoudigInformatieObjectLock> {
```

Surely the UUID is not optional here. :-)